src/pluggy/_manager.py

```diff
@@ -181,7 +181,26 @@ def parse_hookimpl_opts(self, plugin: _Plugin, name: str) -> HookimplOpts | None
         customize how hook implementation are picked up. By default, returns the
         options for items decorated with :class:`HookimplMarker`.
         """
-        method: object = getattr(plugin, name)
+
+        # IMPORTANT: @property methods can have side effects, and are never hookimpl
+        # if attr is a property, skip it in advance
+        plugin_class = plugin if inspect.isclass(plugin) else type(plugin)
+        if isinstance(getattr(plugin_class, name, None), property):
+            return None
+
+        # pydantic model fields are like attrs and also can never be hookimpls
```

Comment:
I am inclined against adding special handling for pydantic . Is this an optimization or does inspecting pydantic classes cause any issues?

Reply:
Well currently it throws an exception if you try to use any pydantic object as a plugin. I would describe it more of an "avoid a known problem with a certain common type of object in the python ecosystem" and less of an optimization, as it's not about speed.

Reply:
If I am understanding the fix correctly, the scenario is that `plugin` is a Pydantic object, the fields (e.g. `hook_type` and `tags` in `BaseCheck`) are returned from `dir(obj)`, but trying to `getattr` them raises `AttributeError`?

Reply:
could this be settled by determining if a static getattr on the type returns a function thats marked?

i agree that we shouldnt fail on a pydantic model, but i wonder if there is a more general way to determine what attributes we can search

Reply:
Ok yeah I'm sure there is some other way to determine incompatible fields besides doing an explicit pydantic check. I'll toy around a bit more and report back.

Reply:
Good news! The explicit pydantic check is actually not needed. I've updated the PR with some improvements and provided an exhaustive test case here: https://gist.github.com/pirate/66f12beac594c99c697cd5543a1cb77b